src/main/controllers/CaseDocumentController.ts

```diff
@@ -30,11 +31,12 @@ export default class CaseDocumentController {
       const userCases = await getUserCasesByLastModified(req);
       const allDocumentSets = combineUserCaseDocuments(userCases);
 
-      const details = allDocumentSets.find(doc => doc.id === docId);
+      const details = allDocumentSets.find(doc => doc !== undefined && doc.id === docId);
```

Comment:
could just check doc && doc.id ===..., unless you want it to throw a NPE if doc is null